alpha/action/merge.go

```diff
@@ -0,0 +1,264 @@
+package action
+
+import (
+	"fmt"
+	"sort"
+
+	"github.com/imdario/mergo"
+
+	"github.com/operator-framework/operator-registry/alpha/declcfg"
+)
+
+// MergeType is the merge strategy used to combine declarative config objects
+// that have the same unique key.
+type MergeType int
+
+const (
+	// PreferLast just uses the last/most recent object.
+	// This is the default merge type.
+	PreferLast MergeType = iota
+	// TwoWay fills in object fields by combining objects in ascending order,
+	// starting with the first/oldest.
+	// Bundle properties are NOT merged because an arbitrary property's key is unknowable.
+	TwoWay
+)
+
+func (mt MergeType) MergeDC(cfg *declcfg.DeclarativeConfig) error {
```

Comment:
This function worries me because it has to be maintained and updated every time we add/remove a merge implementation.

WDYT about refactoring to a `type Merger interface { Merge(*declcfg.DeclarativeConfig) error }` and then implementing each strategy as a different struct implementation of that interface?

Since (AFAICT) this PR doesn't introduce a way to customize the merge implementation via a flag, that would remove this switch entirely from the codebase. If we ever did add a flag, this switch would have to exist in some form, but it would likely be in the CLI, not in the library.

The other benefit is that any consumers of the library could add their own merge implementation without requiring us to merge anything.